docs/source/literal_types.rst

```diff
@@ -289,6 +289,74 @@ using ``isinstance()``:
 This feature is sometimes called "sum types" or "discriminated union types"
 in other programming languages.
 
+Exhaustive checks
+*****************
+
+One may want to check that some code covers all possible ``Literal`` cases, example:
+
+.. code-block:: python
+
+  from typing import Literal
+
+  PossibleValues = Literal['one', 'two']
+
+  def validate(x: PossibleValues) -> bool:
+      if x == 'one':
+          return True
+      elif x == 'two':
+          return False
+      raise ValueError('Wrong values passed: {0}'.format(x))
+
+  assert validate('one') is True
+  assert validate('two') is False
+
+In the code above it is really easy to make a mistake in the future: 
+by adding a new literal value to ``PossibleValues``, 
+but not adding its handler to ``validate`` function:
+
+.. code-block:: python
+
+  PossibleValues = Literal['one', 'two', 'three']
+
+Mypy won't catch that ``'three'`` is not covered.
+However, if you want to have exhaustive check, you need to guard it properly:
+
+.. code-block:: python
+
+  from typing import Literal, NoReturn
+
+  PossibleValues = Literal['one', 'two']
+
+  def assert_exhaustive(value: NoReturn) -> NoReturn:
+      # This also works in runtime as well:
+      assert False, 'This code should never be reached, got: {0}'.format(value)
+
+  def validate(x: PossibleValues) -> bool:
+      if x == 'one':
+          return True
+      elif x == 'two':
+          return False
+      assert_exhaustive(x)
+
+In this case, when adding new values to ``PossibleValues``:
+
+.. code-block:: python
+
+  PossibleValues = Literal['one', 'two', 'three']
+
+Mypy will cover you:
+
+.. code-block:: python
+
+  def validate(x: PossibleValues) -> bool:
+      if x == 'one':
+          return True
+      elif x == 'two':
+          return False
+      assert_exhaustive(x)  # E: Argument 1 to "assert_exhaustive" has incompatible type "Literal['three']"; expected "NoReturn"
+
+This technique works with ``Enum`` values as well.
```

Comment:
I would move the Enum reference to the start, and maybe even make this a default example (while mentioning it also works for literal types with a short example). IMO literal types are more for legacy code, while enums are the future.